### PR TITLE
Make Miri backtraces work with #[global_allocator]

### PR DESCRIPTION
**Pull request by DrMeepster**
Supports changes in rust-lang/miri#1975

### CODE REVIEW

**Review by alexcrichton — COMMENTED** [review]
Seems reasonable to me, but how would you like this to get landed? The CI here is failing presumably because miri needs an update, but does the lack of changes here block a miri update?

File: src/lib.rs [hunk 1]
```diff
@@ -93,6 +93,7 @@
     all(feature = "std", target_env = "sgx", target_vendor = "fortanix"),
     feature(sgx_platform)
 )]
+#![cfg_attr(miri, feature(allocator_api))]
```
[the remark this answers is not in the document]
Comment (DrMeepster) [reply src/lib.rs:96]: Oh whoops, that was from when I was trying to make it use Box<T, System>. I forgot to remove it....

**Comment by DrMeepster** [comment]
> The CI here is failing presumably because miri needs an update, but does the lack of changes here block a miri update?

The new approach should avoid this blocking problem by being backwards compatible with old code

**Comment by DrMeepster** [comment]
Alright, should be updated for the final version of rust-lang/miri#1975

**Comment by DrMeepster** [comment]
But I forgot that this won't pass yet until next nightly with miri on it 🙃

**Comment by RalfJung** [comment]
First the Miri submodule needs to be updated. :) https://github.com/rust-lang/rust/pull/95144 does that.

**Comment by RalfJung** [comment]
@DrMeepster you could try rebasing so that CI picks up the new Miri? Then we'll see if this works as it should.

**Comment by DrMeepster** [comment]
Whoops! I forgot about this completely, sorry.

**Comment by RalfJung** [comment]
Miri CI is green, so LGTM. :)  But I cannot land PRs here.

**Comment by RalfJung** [comment]
:tada: :-)

@alexcrichton what does it take to get this into rustc?

**Pull request merged**